src/codegen/impl_partialeq.rs

```diff
@@ -0,0 +1,122 @@
+
+use ir::comp::{CompInfo, CompKind, Field, FieldMethods};
+use ir::context::BindgenContext;
+use ir::item::Item;
+use ir::ty::{TypeKind, RUST_DERIVE_IN_ARRAY_LIMIT};
+use quote;
+
+pub fn gen_partialeq_impl(ctx: &BindgenContext, comp_info: &CompInfo, item: &Item, ty_for_impl: &quote::Tokens) -> Option<quote::Tokens> {
+    let _ty = item.expect_type();    
+    let mut tokens = vec![];
+
+    if comp_info.kind() == CompKind::Union {
+        // TODO: Is other cases are possible?
+        tokens.push(quote! {
+            &self.bindgen_union_field[..] == &other.bindgen_union_field[..]
+        });
+    } else {
+        for (i, base) in comp_info.base_members().iter().enumerate() {
+            // TODO: Move base field name generation into either in IR or in context.
```

Comment:
@fitzgen There is a similar thing as with anonymous field names.
It's kinda works but ugly. What is the best possible way there?

Reply:
Let's add a `field_name` member to `Base`, compute it upon construction of the `Base`, and then use that name both here and in codegen.

In fact, that could even be its own little PR, and we could land it independently of the rest of this stuff.

Reply:
> In fact, that could even be its own little PR, and we could land it independently of the rest of this stuff.

Done https://github.com/rust-lang-nursery/rust-bindgen/pull/1014 !